projects/media-viewer/src/lib/store/reducers/annotatons.reducer.ts

```diff
@@ -0,0 +1,208 @@
+  state = initialState,
+  action: fromAnnotations.AnnotationsActions
+): AnnotationSetState {
+  switch (action.type) {
+
+    case fromAnnotations.ADD_PAGE: {
+      const payload = action.payload;
+      const  numberOfPages = (state.pages.numberOfPages <= payload.pageNumber) ?
+        payload.pageNumber : state.pages.numberOfPages;
+      const styles = {
+        'left': payload.div['offsetLeft'],
+        'height': payload.div['offsetHeight'],
+        'width': payload.div['offsetWidth']
+      };
+      const scaleRotation = {
+        scale: payload.scale,
+        rotation: payload.rotation
+      };
+
+      const page = {
+        numberOfPages,
+        styles,
+        scaleRotation
+      };
+
+      const pages = {
+        ...page
+      };
+      return {
+        ...state,
+        pages
+      };
+    }
+
+    case fromAnnotations.LOAD_ANNOTATION_SET: {
+      return {
+        ...initialState,
+        loading: true
+      };
+    }
+    case fromAnnotations.LOAD_ANNOTATION_SET_SUCCESS: {
+      const annotationSet = action.payload;
+      const annotationEntities = StoreUtils.generateAnnotationEntities(annotationSet.annotations);
+      const annotationPageEntities = StoreUtils.generatePageEntities(annotationSet.annotations);
+      const commentEntities = StoreUtils.generateCommentsEntities(annotationSet.annotations);
+      return {
+        ...state,
+        annotationSet,
+        annotationEntities,
+        annotationPageEntities,
+        commentEntities,
+        loading: false,
+        loaded: true
+      };
+    }
+
+    case fromAnnotations.SAVE_ANNOTATION_SUCCESS: {
+      const anno = action.payload;
+      const annEntities = {
+        ...state.annotationEntities,
+        [anno.id]: anno
+      };
+      const annotArray = Object.keys(annEntities).map(key => annEntities[key]);
+      const annotationEntities = StoreUtils.generateAnnotationEntities(annotArray);
+      const annotationPageEntities = StoreUtils.generatePageEntities(annotArray);
+      const commentEntities = StoreUtils.generateCommentsEntities(annotArray);
+      const selectedAnnotation = {
+        ...state.selectedAnnotation,
+        annotationId : anno.id,
+        editable: false
+      };
+      return {
+        ...state,
+        annotationEntities,
+        annotationPageEntities,
+        commentEntities,
+        selectedAnnotation,
+        loading: false,
+        loaded: true
+      };
+    }
+
+    case fromAnnotations.DELETE_ANNOTATION_SUCCESS: {
+      const id = action.payload;
+      const page = state.annotationEntities[id].page;
+      const annotationEntities = {
+        ...state.annotationEntities
+      }
+      delete annotationEntities[id];
+      const pageAnnotationsRemoved = [
+        ...state.annotationPageEntities[page].filter(anno => anno.id !== id)
+      ];
+      const annotationPageEntities = {
+        ...state.annotationPageEntities,
+        [page]: pageAnnotationsRemoved
+      };
```

Comment:
this is a bit confusing for me. maybe we can have a walkthrough of the logic in the store section

Reply:
yeap we did